cpp/src/reshape/interleave_columns.cu

```diff
@@ -19,16 +19,102 @@
 #include <cudf/detail/nvtx/ranges.hpp>
 #include <cudf/table/table_device_view.cuh>
 #include <cudf/types.hpp>
+#include <strings/utilities.cuh>
 
 namespace cudf {
 namespace detail {
 namespace {
 struct interleave_columns_functor {
   template <typename T, typename... Args>
-  std::enable_if_t<not cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>> operator()(
-    Args&&... args)
+  std::enable_if_t<not cudf::is_fixed_width<T>() and not std::is_same<T, cudf::string_view>::value,
+                   std::unique_ptr<cudf::column>>
+  operator()(Args&&... args)
   {
-    CUDF_FAIL("Only fixed-width types are supported in interleave_columns.");
+    CUDF_FAIL("interleave_columns not supported for dictionary and list types.");
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_same<T, cudf::string_view>::value, std::unique_ptr<cudf::column>>
+  operator()(table_view const& strings_columns,
+             bool create_mask,
+             rmm::mr::device_memory_resource* mr,
+             cudaStream_t stream = 0)
+  {
+    auto num_columns = strings_columns.num_columns();
+    if (num_columns == 1)  // Single strings column returns a copy
+      return std::make_unique<column>(*(strings_columns.begin()), stream, mr);
+
+    auto strings_count = strings_columns.num_rows();
+    if (strings_count == 0)  // All columns have 0 rows
+      return strings::detail::make_empty_strings_column(mr, stream);
+
+    // Create device views from the strings columns.
+    auto table       = table_device_view::create(strings_columns, stream);
+    auto d_table     = *table;
+    auto num_strings = num_columns * strings_count;
+
+    std::pair<rmm::device_buffer, size_type> valid_mask{{}, 0};
+    if (create_mask) {
+      // Create resulting null mask
+      valid_mask = cudf::detail::valid_if(
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(num_strings),
+        [num_columns, d_table] __device__(size_type idx) {
+          auto cidx = idx % num_columns;
+          auto ridx = idx / num_columns;
+          return !d_table.column(cidx).is_null(ridx);
+        },
+        stream,
+        mr);
+    }
+
+    auto const null_count = valid_mask.second;
+
+    // Build offsets column by computing sizes of each string in the output
+    auto offsets_transformer = [num_columns, d_table] __device__(size_type idx) {
+      // First compute the column and the row this item belongs to
+      auto cidx = idx % num_columns;
+      auto ridx = idx / num_columns;
+      size_type bytes{0};
+      if (!d_table.column(cidx).is_null(ridx))
+        bytes = d_table.column(cidx).element<string_view>(ridx).size_bytes();
+      return bytes;
```

Comment:
Is it possible to simplify this?
```suggestion
      return d_table.column(cidx).element<string_view>(ridx).size_bytes();
```

Reply:
no - the pre-condition to invoking the `element` method is that the row has a valid item.

Reply:
> no - the pre-condition to invoking the element method is that the row has a valid item.

Not quite. You can invoke `element<T>(i)` if `i` is null, but it's UB to _use_ the returned object. 

```
if(col.is_null(i)){
   auto element = col.element<T>(i); // This is fine
   auto other = element; // This is UB
   if(element > 0) // this is UB
   // etc. Any attempt to _read_ from `element` is UB
}
```

Reply:
Also, this a perfect example of @codereport's Initialize-Then-Modify anti-pattern.

Instead, do:
```
return d_table.column(cidx).is_valid(ridx) ? d_table.column(cidx).element<string_view>(ridx).size_bytes() : 0;
```

Reply:
ok - i should have said "...pre-condition to *using* the element..."

Reply:
Specifically for string views, though, is accessing bytes UB? If I recall, it’s always initialized.